### PR TITLE
chore(qa): Set up testing and linting

### PR DESCRIPTION
**Pull request by petermasking**
What I've done:

* Set up testing
* Set up linting
* Set up GitHub Actions
   * Dependabot
   * CodeQL workflow
   * Node.js CI workflow (build, test, lint)

### CODE REVIEW

File: tsconfig.json
Comment (BasMasking) [inline tsconfig.json:1]: Why have you included the test folder to the include path? Vitest automatically throws an error when running a test that itsn't valid typescript.

The result of this change is a different structure in the dist folder. The server cannot start as the jitar.js file it is looking for is located in the `dist/src/` in stead of the `dist/` folder it expects now.

**Review by BasMasking — CHANGES_REQUESTED** [review]

File: vite.config.js
Comment (BasMasking) [inline vite.config.js:1]: The vite config is already present as a .ts file. I expect that the build step automatically transpiles this file into a .js file as well.

I don't expect this .js file to be here. Could you check the build configuration and remove the .js file?

**Review by BasMasking — CHANGES_REQUESTED** [review]
I've added a comment about the vite.config.js file, which is probably generated during the build step. This file is already present as a .ts file.

File: vite.config.js
Comment (petermasking) [reply vite.config.js:1]: Great find! While playing around with the new setup, .js files where accidentally generated in the src folder. I've removed most of them, but missed this one.

**Review by BasMasking — APPROVED** [review]

**Pull request merged**